packages/cardpay-reward-programs/cardpay_reward_programs/main.py

```diff
@@ -49,11 +35,11 @@ def run_reward_program(
     """
     with open(AnyPath(parameters_file), "r") as stream:
         parameters = json.load(stream)
-    rule = to_rule(parameters)
-    start_block = rule.start_block
-    end_block = rule.end_block
-    results_df = rule.run(start_block, end_block)
```

Comment:
The start block and end block here would be the valid range of the rule, not the range for one payment cycle - and how to pick the range should be an internal decision in the rule.

Reply:
I **should** have put this as `payment_cycle`

Reply:
We fork here. 

`run(start_block, end_block)` vs `run(payment_cycle, reward_program_id)`

I always thought placing `payment_cycle` and letting the rule infer the range based off the `payment_cycle` will cause us to need to update the rule code when `payment_cycle` are conflicting across rules.

Reply:
Yeah I think this is the main difference - I'm not sure what the conflict would be there, let's talk that through on Monday.